### PR TITLE
feat(notifications): add apple push notifications

### PR DESCRIPTION
**Pull request by iampaulanca**
### Description
Added apple push notifications as one of the many ways of notifying user.

### Testing
Enter the following info in dotenv file. Fill with your own info

APNSAUTHKEY="path to authkey on users machine"
APNSBUNDLEID="app bundle id"
APNSDEVICETOKEN="device token"
APNSKEYID="developers key id"
APNSPRODUCTION="true/false for production"
APNSTEAMID="developers team id"

Then run test with command "npm run test:notification"

expect similar to:

Paul's Macbook$ npm run test:notification

> test:notification
> npm run compile && node build/test/functional/test-notification.js

> compile
> tsc

ＳＴＲＥＥＴＭＥＲＣＨＡＮＴ
3.4.1

[2:20:47 PM] info :: ✔ push notification sent
[2:20:49 PM] info :: ✔ sms sent
[2:20:49 PM] info :: ✔ email sent

### New dependencies
apn@2.2.0

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -423,7 +431,7 @@ const store = {
     'xboxss',
     'xboxsx',
   ]),
-  stores: envOrArray(process.env.STORES, ['amazon', 'bestbuy']).map(entry => {
+  stores: envOrArray(process.env.STORES, ['amazon', 'bestbuy', 'newegg', 'nvidia-gb', 'memoryexpress', 'canadacomputers']).map(entry => {
```
Comment (jef) [inline src/config.ts:434]:
```suggestion
  stores: envOrArray(process.env.STORES, ['amazon', 'bestbuy']).map(entry => {
```

I'd like to keep this slim. It's mostly just for example and to show the app works when first booted.

File: src/config.ts [hunk 2]
```diff
@@ -205,6 +205,14 @@ const logLevel = envOrString(process.env.LOG_LEVEL, 'info');
 
 const notifications = {
   desktop: process.env.DESKTOP_NOTIFICATIONS === 'true',
+  apns: {
+    apnsAuthKey: envOrString(process.env.APNSAUTHKEY),
+    apnsKeyId: envOrString(process.env.APNSKEYID),
+    apnsTeamId: envOrString(process.env.APNSTEAMID),
+    apnsProduction: envOrBoolean(process.env.APNSPRODUCTION),
+    apnsDeviceToken: envOrString(process.env.APNSDEVICETOKEN),
+    apnsBundleId: envOrString(process.env.APNSBUNDLEID)
```
Comment (jef) [inline src/config.ts:214]:
```suggestion
    apnsAuthKey: envOrString(process.env.APNS_AUTHKEY),
    apnsKeyId: envOrString(process.env.APNS_KEYID),
    apnsTeamId: envOrString(process.env.APNS_TEAMID),
    apnsProduction: envOrBoolean(process.env.APNS_PRODUCTION),
    apnsDeviceToken: envOrString(process.env.APNS_DEVICETOKEN),
    apnsBundleId: envOrString(process.env.APNS_BUNDLEID)
```

Do you mind using the `APNS_` prefix?

File: src/notification/apns.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      key: apns.apnsAuthKey,
+      keyId: apns.apnsKeyId,
+      teamId: apns.apnsTeamId
+    },
+    production: apns.apnsProduction
+  };
+   
+  var apnProvider = new apn.Provider(options);
+
+  let deviceToken = apns.apnsDeviceToken;
```
Comment (jef) [inline src/notification/apns.ts:20]:
```suggestion
```

File: src/notification/apns.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  var note = new apn.Notification();
+ 
+  note.badge = 1;
+  note.sound = "ping.aiff";
+  note.alert = "\uD83D\uDCE7 \u2709 You have a new message";
+  note.payload = {'label': '1'};
+  note.topic = apns.apnsBundleId;
+
+  apnProvider.send(note, deviceToken).then( (result) => {
```
Comment (jef) [inline src/notification/apns.ts:30]:
```suggestion
  apnProvider.send(note, apns.apnsDeviceToken).then( (result) => {
```

File: src/notification/apns.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import {Link, Store} from '../store/model';
+import {Print, logger} from '../logger';
+import * as apn from 'apn';
+import {config} from '../config';
+
+const { apns } = config.notifications;
+
+export function sendApns(link: Link, store: Store) {
+  var options = {
```
Comment (jef) [inline src/notification/apns.ts:9]:
```suggestion
  const options = {
```

Do you mind changing this to `const` and a couple variables that don't change below?

**Review by jef — COMMENTED** [review]
Overall looks pretty good! Thanks for the contribution.

Just a few nits. Can you also update `docs/` to show that this is supported?

Thank you!

File: src/config.ts (hunk 1 above)
Comment (iampaulanca) [reply src/config.ts]: Done

File: src/config.ts (hunk 2 above)
Comment (iampaulanca) [reply src/config.ts]: Done

File: src/notification/apns.ts (hunk 3 above)
Comment (iampaulanca) [reply src/notification/apns.ts]: Done

File: src/notification/apns.ts (hunk 4 above)
Comment (iampaulanca) [reply src/notification/apns.ts]: Done

File: src/notification/apns.ts (hunk 5 above)
Comment (iampaulanca) [reply src/notification/apns.ts]: Done

**Comment by jef** [comment]
Do you mind merging into `main` and resolve the package-lock.json conflict? It seems like this branch is far behind.

**Review by jef — APPROVED** [review]
Great!

Thank you very much. Sorry this took awhile for me to get in 😅

**Pull request merged**